player/lua/stats.lua

```diff
@@ -721,114 +736,196 @@ local function pq_eotf(x)
-    else
-        append_property(s, "estimated-vf-fps",
-                        {prefix="FPS:", suffix=" (estimated)"})
-    end
-
     append_display_sync(s)
     append_perfdata(s, o.print_perfdata_passes)
 
-    append_resolution(s, r, "Native Resolution:")
-    if ro and (r["w"] ~= ro["dw"] or r["h"] ~= ro["dh"]) then
-        append_resolution(s, ro, "Output Resolution:", "dw", "dh")
+    if mp.get_property_native("deinterlace") then
+        append_property(s, "deinterlace", {prefix="Deinterlacing:"})
     end
+
     local scale = nil
     if not mp.get_property_native("fullscreen") then
         scale = mp.get_property_native("current-window-scale")
     end
-    append_resolution(s, {w=scaled_width, h=scaled_height, s=scale}, "Scaled Resolution:")
 
-    if mp.get_property_native("deinterlace") then
-        append_property(s, "deinterlace", {prefix="Deinterlacing:"})
+    local r = mp.get_property_native("video-target-params")
+    if not r then
+        local osd_dims = mp.get_property_native("osd-dimensions")
+        local scaled_width = osd_dims["w"] - osd_dims["ml"] - osd_dims["mr"]
+        local scaled_height = osd_dims["h"] - osd_dims["mt"] - osd_dims["mb"]
+        append_resolution(s, {w=scaled_width, h=scaled_height, s=scale},
+                          "Resolution:")
+        return
     end
 
-    append(s, r["pixelformat"], {prefix="Pixel Format:"})
-    if r["hw-pixelformat"] ~= nil then
-        append(s, r["hw-pixelformat"], {prefix_sep="[", nl="", indent=" ",
-                suffix="]"})
+    -- Add window scale
+    r["s"] = scale
+
+    append_img_params(s, r)
+    append_hdr(s, r, true)
+end
+
+
+local function add_video(s)
+    local r = mp.get_property_native("video-params")
+    local ro = mp.get_property_native("video-out-params")
+    -- in case of e.g. lavfi-complex there can be no input video, only output
+    if not r then
+        r = ro
+    end
+    if not r then
+        return
     end
-    append(s, r["colorlevels"], {prefix="Levels:", nl=""})
 
-    -- Group these together to save vertical space
-    append(s, r["colormatrix"], {prefix="Colormatrix:"})
-    append(s, r["primaries"], {prefix="Primaries:", nl=""})
-    append(s, r["gamma"], {prefix="Transfer:", nl=""})
+    local osd_dims = mp.get_property_native("osd-dimensions")
+    local scaled_width = osd_dims["w"] - osd_dims["ml"] - osd_dims["mr"]
+    local scaled_height = osd_dims["h"] - osd_dims["mt"] - osd_dims["mb"]
+
+    append(s, "", {prefix=o.nl .. o.nl .. "Video:", nl="", indent=""})
+    if append_property(s, "video-codec", {prefix_sep="", nl="", indent=""}) then
+        append_property(s, "hwdec-current", {prefix="HW:", nl="", indent=o.prefix_sep,
+                         no_prefix_markup=false, suffix=""}, {no=true, [""]=true})
+    end
+    local has_prefix = false
+    if append_property(s, "estimated-frame-number", {prefix="Frame:"}) then
+        append_property(s, "estimated-frame-count", {indent=" / ", nl="",
```

Comment:
mpv has no accurate frame numbers, this should never be displayed because users WILL rely on it

Reply:
I agree. It was request from @arm64-v9a, what do you think?

Reply:
For me even close frame number is good enough to know where to look at with other tools. Also it should be pretty accurate for sane video.

But I agree if the worry is that it is confusing for users, we can remove it.

Reply:
> mpv has no accurate frame numbers, this should never be displayed because users WILL rely on it

But which player has accurate frame numbes? 🤔

Reply:
@sfan5: How about I add option to stats.lua to show this estimated frame numbers? Disabled by default. If users enable it, it is their fault.

Reply:
That sounds good to me

Reply:
Option + disabled by default is fine by me.

Reply:
> That sounds good to me, that option should also include all the other frame information such as frame type and interlacing

The main issue are estimated parameters, others aren't. But sure I can skip it, just displaying "B Interlaced" is not that useful. With frame number it makes whole package.